GlobalBehaviorandComponents/managetrustedorigins.py

```diff
@@ -0,0 +1,113 @@
+    user_info = get_userinfo()
+    okta_admin = OktaAdmin(session[SESSION_INSTANCE_SETTINGS_KEY])
+
+    trustedorigins = okta_admin.get_trusted_origin()
+    trustedoriginslist = []
+
+    for trustedorigin in trustedorigins:
+        if user_info["sub"] in trustedorigin["name"]:
+            trustedoriginslist.append(trustedorigin)
+
+    return render_template(
+        "/managetrustedorigins.html",
+        templatename=get_app_vertical(),
+        user_info=get_userinfo(),
+        config=session[SESSION_INSTANCE_SETTINGS_KEY],
+        trustedorigins=trustedoriginslist)
+
+
+@gbac_managetrustedorigins_bp.route("/edittrustedorigins")
+@apply_remote_config
+@is_authenticated
+def gbac_trustedorigins_edit():
+    logger.debug("gbac_trustedorigins_edit()")
+    # user_info = get_userinfo()
+    trustedoriginid = request.args.get('trustedoriginid')
+
+    if trustedoriginid:
+        okta_admin = OktaAdmin(session[SESSION_INSTANCE_SETTINGS_KEY])
+        trustedorigin = okta_admin.get_trusted_origin(id=trustedoriginid)
+
+        return render_template(
+            "/managetrustedoriginscreateupdate.html",
+            templatename=get_app_vertical(),
+            user_info=get_userinfo(),
+            config=session[SESSION_INSTANCE_SETTINGS_KEY],
+            trustedorigin=trustedorigin)
+    else:
+        return redirect(url_for(
+            "gbac_managetrustedorigins_bp.gbac_trustedorigins",
+            _external=True,
+            _scheme=session[SESSION_INSTANCE_SETTINGS_KEY]["app_scheme"]))
+
+
+@gbac_managetrustedorigins_bp.route("/createtrustedorigins")
+@apply_remote_config
+@is_authenticated
+def gbac_trustedorigins_create():
+    logger.debug("gbac_trustedorigins_create()")
+
+    return render_template(
+        "/managetrustedoriginscreateupdate.html",
+        templatename=get_app_vertical(),
+        user_info=get_userinfo(),
+        config=session[SESSION_INSTANCE_SETTINGS_KEY])
+
+
+@gbac_managetrustedorigins_bp.route("/deletetrustedorigins")
+@apply_remote_config
+@is_authenticated
+def gbac_trustedorigins_delete():
+    logger.debug("gbac_trustedorigins_delete()")
+    okta_admin = OktaAdmin(session[SESSION_INSTANCE_SETTINGS_KEY])
+    trustedoriginid = request.args.get('trustedoriginid')
+    okta_admin.delete_trusted_origin(trustedoriginid)
+    message = "Trusted Origin Deleted"
+    return redirect(url_for(
+        "gbac_managetrustedorigins_bp.gbac_trustedorigins",
+        _external=True,
+        _scheme=session[SESSION_INSTANCE_SETTINGS_KEY]["app_scheme"],
+        message=message))
+
+
+@gbac_managetrustedorigins_bp.route("/createtrustedoriginsfinal")
+@apply_remote_config
+@is_authenticated
+def gbac_trustedorigins_create_final():
+    logger.debug("gbac_trustedorigins_create_final()")
+    user_info = get_userinfo()
+    okta_admin = OktaAdmin(session[SESSION_INSTANCE_SETTINGS_KEY])
+    trustedoriginurl = request.args.get('trustedoriginurl')
+    trustedoriginid = request.args.get('id')
```

Comment:
Not checking if variables exist. What happens when it `None` or empty?

Reply:
Added check and send http error code if empty.